pkg/registry/populator.go

```diff
@@ -379,18 +380,43 @@ func (i *DirectoryPopulator) loadOperatorBundle(manifest PackageManifest, bundle
 	return nil
 }
 
+type bundleVersion struct {
+	name    string
+	version semver.Version
+
+	// Keep track of the number of times we visit each version so we can tell if a head is contested
+	count int
+}
+
+// compare returns a value less than one if the receiver arg is less smaller the given version, greater than one if it is larger, and zero if they are equal.
+// This comparison follows typical semver precedence rules, with one addition: a simple ordering is defined for build metadata (+<build-metadata>) corresponding to whole numbers such that they are in ascending order; e.g. 1.0.0 < 1.0.0+1 < 1.0.0+2 ...
+func (b bundleVersion) compare(v bundleVersion) int {
+	if c := b.version.Compare(v.version); c != 0 {
+		return c
+	}
+
+	// Attempt to compare based on build metadata
+	return buildInt(b.version).Cmp(buildInt(v.version))
```

Comment:
this comparison needs to be more sophisticated.  we should follow the logic semver uses for comparing prereleases, which as i understand it is to break the prerelease into dotted segments, and compare each segment until it finds a difference.

Perhaps you can take the buildID, and append it to a "v1.0.0" version as a prerelease, and do that comparison using semver libs, to determine which buildID is greater.

e.g.:  v1.2.3+1.2.3 vs v.1.2.3+1.1.55, you extract the buildid as 1.2.3 and 1.1.55, then do a semver comparison of "v1.0.0-1.2.3" vs "v1.0.0-1.1.55", and whichever one of those is greater, means that buildID is greater.

Reply:
sgtm -- I'll prototype it out.

Reply:
I think we can make this work for most cases by:

- extract buildId
- make a new semver string (or build it directly), but set major/minor/patch to `0` and set `prerelease` to the buildid
- compare both of those constructed versions

That is basically saying "apply the prerelease comparison rules to buildid" which I think covers a lot of cases (when buildid is not valid semver, etc). 

did a quick poc in the playground: https://play.golang.org/p/jMDNbBlP_Ib